DCOS/fluentd-testing/preprovision-agent-windows.ps1

```diff
@@ -0,0 +1,60 @@
+$here = Split-Path -Parent $MyInvocation.MyCommand.Path
+$TD_AGENT = "td-agent-3.1.1-0-x64.msi"
+$url = "http://packages.treasuredata.com.s3.amazonaws.com/3/windows/$TD_AGENT"
+$MSI_PATH = Join-Path $here $TD_AGENT
+$fluentPath = "$env:SystemDrive\opt\td-agent\embedded\bin"
+$env:Path = "$fluentPath;" + $env:Path.Replace("$fluentPath;", "")
+
+function Get-Fluentd () {
+    Write-Host "Downloading fluentd/td-agent..."
+    $wc = New-Object System.Net.WebClient
+    $wc.DownloadFile($url, $MSI_PATH)
+}
+
+function Install-Fluentd () {
+    Write-Host "Installing fluentd..."
+    
+    [System.IO.FileSystemInfo] $file = Get-Item $MSI_PATH
+    [String] $DataStamp = get-date -Format yyyyMMddTHHmmss
+    [String] $logFile = '{0}-{1}.log' -f $file.FullName, $DataStamp
+    
+    $MSIArguments = @(
+        "/i"                         # Install
+        ('"{0}"' -f $file.fullname)  # MSI path
+        "/qn"                        # No UI
+        "/norestart"                 # No computer restart
+        "/L*v"                       # Log everything including verbose
+        $logFile                     # Log file name
+    )
+    
+    $process = Start-Process "msiexec.exe" -ArgumentList $MSIArguments -Wait -NoNewWindow -PassThru
+    if ($process.ExitCode -ne 0) {
+        throw "ERROR: Failed to install Fluentd. Error code: " + $process.ExitCode + " See details in log: $logfile"
+    }
+}
+
+function Register-FluentdService () {
+    Write-Host "Registering fluentd service..."
+    fluentd --reg-winsvc i
+    fluentd --reg-winsvc-fluentdopt "-c $env:SystemDrive/opt/td-agent/etc/td-agent/td-agent.conf -o $env:SystemDrive/opt/td-agent/td-agent.log"
+}
+
+function Start-FluentdService () {
+    Write-Host "Starting fluentd service..."
+    Start-Service fluentdwinsvc
+    
+    Write-Output '{"message":"hello from powershell"}' | fluent-cat debug.event
```

Comment:
We're calling `fluent-cat` which I assume is a `exe` binary from the disk. Same question as above: Is it added to the `$env:PATH` at this point ? Also, please add a check for `$LASTEXITCODE`